README.md

```diff
@@ -29,9 +29,194 @@ This library is written in modern JavaScript and is published in both CommonJS a
+    publicKey,  // The generated public key composed in a specific format. 
+}
+```
+
+#### Parameters
+
+<details><summary><strong>mnemonic</strong></summary>
+
+Type: `String`
+
+The mnemonic provided as one of the recovery methods for a key pair.
+</details>
+
+[algorithm](#algorithm)
+
+[options](#options)
+
+### getKeyPairFromSeed(seed, algorithm, options)
+
+Generates a key pair based on the specified seed and algorithm.
+
+Returns an object with the following:
+```js
+{
+    privateKey, // The generated private key composed in a specific format.
+    publicKey,  // The generated public key composed in a specific format. 
+}
 ```
 
+#### Parameters
+
+<details><summary><strong>seed</strong></summary>
+
+Type: `String`
+
+The seed provided as one of the recovery methods for a key pair.
+</details>
+
+[algorithm](#algorithm)
+
+[options](#options)
+
+### Common Parameters
+<details name="algorithm"><summary><strong>algorithm</strong></summary>
+ 
+Type: `Object` or `String`
+
+The algorithm identifier and the respective parameters to generate a key pair.
+
+It can be specified as an `Object` or a `String`. Using an `Object` will provide freedom to override default algorithm parameters in relation to its type. On the other hand, a `String` presents a useful and quick approach if the default parameters are suitable.
+
+The default parameters are different for each algorithm type. Currently only 2 types are supported:
+
+<details><summary><strong>RSA</strong></summary>
+
+Default Parameters:
+```js
+modulusLength: 2048		    // Number
+publicExponent: 65537		// Number
+method: 'PRIMEINC'		    // String
+```
+
+You can override only the parameters that you need, all the other ones remain with default values. 
+
+> ⚠️ Please make sure that values follow the same type as default ones. Also, parameters that are not available as default are not supported.
+
+Example `Object`:
+```js
+const algorithm = {
+    id: 'rsa',
+    modulusLength: 4096,
+}
+```
+
+Example `String`:
+```js
+const algorithm = 'rsa';
+```
+
+In the examples above we are using an alias for RSA encryption. Although this is possible, the full list of supported RSA key algorithms can be found in the [RSA Keys Section](https://github.com/ipfs-shipyard/js-crypto-key-composer/tree/initial-impl#key-algorithms) of [crypto-key-composer](https://github.com/ipfs-shipyard/js-crypto-key-composer) package.
+</details>
+
+<details><summary><strong>ED25519</strong></summary>
+
+This algorithm doesn't have any default parameters since it only needs a 32 bytes seed.
+
+Example `Object`:
+```js
+const algorithm = { id: 'ed25519' };
+```
+
+Example `String`:
+```js
+const algorithm = 'ed25519';
+```
+</details>
+
+----------
+</details>
+
+<details name="options"><summary><strong>options</strong></summary>
+
+Type: `Object`
+
+Currently, options are available for composability purposes. Your key pair will be composed and, if needed, encrypted based on these. 
```

Comment:
Maybe: The current options allow you to decide on the key format, the key encryption and the password to use to encrypt the key.